src/components/context/DataSyncContext.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+export enum SyncStatus {
+  NO_CHANGES = 'noChanges',
+  SYNCHRONIZED = 'synchronized',
+  ERROR = 'error',
+  SYNCHRONIZING = 'synchronizing',
+}
+
+type ComputeStatusParam = {
+  isLoading: boolean;
+  isSuccess: boolean;
+  isError: boolean;
+};
+
+type DataSyncContextType = {
+  status: SyncStatus;
+  computeStatusFor: (
+    requestKey: string,
+    requestStatus: ComputeStatusParam,
+  ) => void;
+};
+
+const DataSyncContext = createContext<DataSyncContextType>({
+  status: SyncStatus.NO_CHANGES,
+  computeStatusFor: (_requestKey, _requestStatus) => {
+    console.error(
+      'No Provider found for "DataSyncContext". Check that this Provider is accessible.',
+    );
+  },
+});
+
+type Props = {
+  children: JSX.Element | JSX.Element[];
+};
+
+const containsStatus = (allStatus: SyncStatus[], status: SyncStatus) =>
+  allStatus.some((s) => s === status);
+
+export const DataSyncContextProvider = ({ children }: Props): JSX.Element => {
```

Comment:
It is the global state for publication right? It looks like really generic, which is okay. But maybe adding comments would be good so it's easily reusable!

Reply:
I can try to find a better name. For now it is the global publication's state but we could reuse it for every thing we want to indicate the synchronisation state to the user if we would like.